src/models/user.ts

```diff
@@ -3,21 +3,28 @@ import { Document } from 'dynamoose/dist/Document';
 import Semester, { SemesterInterface } from './semester';
 
 export const userSchema = new dynamoose.Schema(
-    {
-        id: String,
-        semesters: {
-            type: Array,
-            schema: Semester,
-        },
-    },
-    {
-        timestamps: true,
-    }
+	{
+		id: {
+			type: String,
+			hashKey: true,
+		},
+		semesters: {
+			type: Set,
+			schema: [Semester],
+		},
+		owner: {
+			type: String,
+		},
+	},
+	{
+		timestamps: true,
+	}
 );
 
 export interface UserInterface extends Document {
-    owner: string;
-    semesters: SemesterInterface[];
+	id: string;
+	owner: string;
+	semesters: SemesterInterface[];
```

Comment:
Should this be a `Set<SemesterInterface>` instead of an array? 
I did some searching to try to figure out why I used Set for the SemesterInterface's courses prop, but I can't find anything. Maybe to ensure each item (semester in this case) is unique?

Reply:
IIRC we also were required to use sets?